geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/VisibilitiesTest.scala

```diff
@@ -8,224 +8,94 @@
-
-    // write some data
-    val fs = ds.getFeatureSource(sftName)
-
-    val features = getFeatures(sft).toList
-    val privFeatures = features.take(3)
-    privFeatures.foreach { f => f.getUserData.put(SecurityUtils.FEATURE_VISIBILITY, "user&admin") }
-
-    val nonPrivFeatures = features.drop(3)
-    nonPrivFeatures.foreach { f => f.getUserData.put(SecurityUtils.FEATURE_VISIBILITY, "user") }
-
-    fs.addFeatures(new ListFeatureCollection(sft, privFeatures ++ nonPrivFeatures))
-    fs.flush()
-
-    val ff = CommonFactoryFinder.getFilterFactory2
-    import ff.{literal => lit, property => prop, _}
-
-    val unprivDS = DataStoreFinder.getDataStore(Map(
-      "instanceId"        -> instanceId,
-      "zookeepers"        -> "zoo1:2181,zoo2:2181,zoo3:2181",
-      "user"              -> "nonpriv",
-      "password"          -> "nonpriv",
-      "tableName"         -> "testwrite",
-      "useMock"           -> "true",
-      "featureEncoding"   -> "avro")).asInstanceOf[AccumuloDataStore]
-
-    "priv should be able to delete a feature" in {
-      fs.removeFeatures(ff.id(new FeatureIdImpl("1")))
-      fs.flush()
 
-      "using ALL queries" in {
-        SelfClosingIterator(fs.getFeatures(new Query(sftName)).features) must haveLength(5)
-      }
-
-      "using record id queries" in {
-        fs.getFeatures(ff.id(ff.featureId("1"))).features().hasNext must beFalse
-      }
-
-      "using attribute queries" in {
-        val filter = or(
-          ff.equals(prop("name"), lit("1")),
-          ff.equals(prop("name"), lit("4")))
-
-        val reader = ds.getFeatureReader(new Query(sftName, filter), Transaction.AUTO_COMMIT)
-        SelfClosingIterator(reader) must haveLength(1)
+    "keep unprivileged from deleting secured features" in {
+      unprivDS.getFeatureSource(sftName).asInstanceOf[SimpleFeatureStore].removeFeatures(ECQL.toFilter("IN('2')"))
+      foreach(filters) { filter =>
+        val reader = privDS.getFeatureReader(new Query(sftName, filter), Transaction.AUTO_COMMIT)
+        SelfClosingIterator(reader).toList must containTheSameElementsAs(privFeatures ++ unprivFeatures)
       }
     }
 
-    "nonpriv should not be able to delete a priv feature" in {
-      val unprivFS = unprivDS.getFeatureSource(sftName).asInstanceOf[SimpleFeatureStore]
-      unprivFS.removeFeatures(ff.id(new FeatureIdImpl("2")))
-      unprivFS.flush()
-
-      "priv should still see the feature that was attempted to be deleted" in {
-        fs.getFeatures(ff.id(ff.featureId("2"))).features().hasNext must beTrue
+    "allow privileged to delete secured features" in {
+      privDS.getFeatureSource(sftName).asInstanceOf[SimpleFeatureStore].removeFeatures(ECQL.toFilter("IN('2')"))
+      foreach(filters) { filter =>
+        val reader = privDS.getFeatureReader(new Query(sftName, filter), Transaction.AUTO_COMMIT)
+        SelfClosingIterator(reader).toList must containTheSameElementsAs(privFeatures.take(2) ++ unprivFeatures)
       }
     }
   }
```

Comment:
Sanity check:  this was just refactored?  I.e., no lose of test coverage?

Reply:
yes should still cover everything (actually more I think since it checks each type of filter now)